src/filesystem/nodes/node_data.rs

```diff
@@ -78,10 +79,14 @@ impl NodeData {
             }
             NodeData::File {
                 permissions,
+                associated_size,
                 associated_data,
                 content,
             } => {
                 written_bytes += permissions.encode(writer).await?;
+                let associated_bytes = associated_size.to_le_bytes();
+                writer.write_all(&associated_bytes).await?;
+                written_bytes += associated_bytes.len();
```

Comment:
This change is unfortunately format breaking and we can't include it in this PR. It's worth doing but we now have production data in our current format and need to bump the format version and write migrations to handle these kind of breaks.

Reply:
Makes sense, I can move this change to another branch to deal with separately